src/main/java/dylanbot/Task.java

```diff
@@ -0,0 +1,57 @@
+package dylanbot;
+
+/**
+ * Represents a general Task
+ */
+public class Task {
```

Comment:
Perhaps you can make this abstract? I think this will prevents errors when implementing since Task does not have a type